### ISSUE TITLE
TS def error with moment

### ISSUE THREAD

**Issue by trungk18** [fullcalendar/fullcalendar#4680]
Hi,

I am trying to format the slotLabelFormat. Currently I am passing the value to the calendar option as such

```javascript
  slotLabelFormat = [
    { weekday: 'short', day: '2-digit', month: '2-digit', omitCommas: true },
    { hour: 'numeric', minute: '2-digit', omitZeroMinute: true, meridiem: 'short' }
  ]
```

https://codepen.io/trungk18/pen/JqPpyr

![image](https://user-images.githubusercontent.com/6767322/57615188-b223e680-75ad-11e9-84dc-c275c6fa1119.png)

It displays as `Sat 3/16` which looks like the US locale. I try to pass locale `en-SG` and it displayed correct as `Sat 16/3` but what I want to display is `Sat, 16 March`.

I couldn't find the documentation for that. Am I missing something?

**Comment by acerix** [comment]
You can use `month: 'long'` to get the month name instead of just the number:

https://codepen.io/anon/pen/gJwygX?editors=0010

This page explains the formatting options:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

If you need more control over the format you can use a plugin:

https://fullcalendar.io/docs/moment-plugins

**Comment by trungk18** [comment]
Thanks acerix,

I saw that the titleFormat can accept string or the callback function
```
titleFormat: 'dddd, MMMM D, YYYY'
```

```
  titleFormat: function(date) {
    return date.toString() + '!!!';
  }
```

Can I instead pass a callback function somewhere in slotLabelFormat for the weekday only, as below? Sorry I still don't get how we could use the moment plugin for that slotLabelFormat.

```
slotLabelFormat = [
    {
        weekday: 'short', day: '2-digit', month: '2-digit', omitCommas: true, callback: function (xx) {
            
        }
    },
    { hour: 'numeric', minute: '2-digit', omitZeroMinute: true, meridiem: 'short' }
]
```

**Comment by trungk18** [comment]
Thanks @acerix ,

I saw the titleFormat could accept either a string, object or a function

```
titleFormat: 'MMMM D, YYYY'
```
```
  titleFormat: function(date) {
    return date.toString() + '!!!';
  }
```

https://fullcalendar.io/docs/slotLabelFormat

But I couldn't image how it will fit in the slotLabelFormat. I am using an array of object to create multiple row. Can I pass a callback like that?

```
  slotLabelFormat = [
    { weekday: 'short', day: '2-digit', month: '2-digit', omitCommas: true, callback: function (params) {
        
    } },
    { hour: 'numeric', minute: '2-digit', omitZeroMinute: true, meridiem: 'short' }
  ]
```

**Comment by acerix** [comment]
If you want to use the "MMMM D, YYYY" format, you need the moment plugin.  `slotLabelFormat` accepts an array where is element is a row, there's anan example with moment.js on the page for version 3:

https://fullcalendar.io/docs/v3/slotLabelFormat

**Comment by trungk18** [comment]
Thanks @acerix,

I am having a hard time fighting with moment plugin distributed via npm. I am using angular.

package.json

```
  "dependencies": {
    "@fullcalendar/angular": "^4.1.0",
    "@fullcalendar/core": "^4.1.0",
    "@fullcalendar/daygrid": "^4.0.0",
    "@fullcalendar/interaction": "^4.1.0",
    "@fullcalendar/list": "^4.1.0",
    "@fullcalendar/moment": "^4.1.0",
```

I imported it in my component

```
import interactionPlugin from "@fullcalendar/interaction";
import resourceTimelinePlugin from "@fullcalendar/resource-timeline";
import momentPlugin from "@fullcalendar/moment";
```

There is an error

![image](https://user-images.githubusercontent.com/6767322/57848125-c44e9080-780a-11e9-8c87-aff4f10a26f4.png)

![image](https://user-images.githubusercontent.com/6767322/57848219-efd17b00-780a-11e9-9309-e1b5a97127c2.png)

You have any idea?

**Comment by acerix** [comment]
The error seems to indicate that it *should* be included like `import * as moment  from "moment";` and I also get the same error so I created an issue for that:

https://github.com/fullcalendar/fullcalendar/issues/4691
↑
Could you please visit the link, **give a** 👍  **to vote** ([?](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments)), press the 🔈**Subscribe** button to receive updates?

For now, I think you need to patch it like you suggested.

**Comment by trungk18** [comment]
Sure, thank you.

**Comment by arshaw** [comment]
fixed in master. will release soon

**Comment by arshaw** [comment]
This was fixed in 4.3.0 though please start using 4.3.1
Releases: https://github.com/fullcalendar/fullcalendar/releases
